mycroft/skills/intent_service.py

```diff
@@ -103,12 +106,25 @@ def __init__(self, bus):
         # Converse method
         self.bus.on('mycroft.speech.recognition.unknown', self.reset_converse)
         self.bus.on('mycroft.skills.loaded', self.update_skill_name_dict)
+        self.bus.on('mycroft.skills.list', self.update_skill_list)
+
 
         def add_active_skill_handler(message):
-            self.add_active_skill(message.data['skill_id'])
+            category = 'undefined'
+            if message.data.get("skill_cat", None) is not None:
+                category = message.data['skill_cat']
+
+            self.add_active_skill(message.data['skill_id'], category)
+
+        def remove_active_skill_handler(message):
+            self.remove_active_skill(message.data['skill_id'])
+            LOG.error("IntentSvc:After remove_skill %s, active_skills=%s" % (message.data['skill_id'], self.active_skills))
 
         self.bus.on('active_skill_request', add_active_skill_handler)
+        self.bus.on('deactivate_skill_request', remove_active_skill_handler)
         self.active_skills = []  # [skill_id , timestamp]
+
+        # BUG fix this!
```

Comment:
Could you elaborate, in case someone else decides to fix it?

Reply:
Its actually not a bug its just a poor design. i removed the comment